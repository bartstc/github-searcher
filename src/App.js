import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRepos } from './modules/fetchedRepos/fetchedActions';
import './App.css';

import SearchForm from './components/SearchForm';
import ReposList from './components/ReposList';
import SavedRepos from './components/SavedRepos';

export class App extends Component {
  state = {
    query: '',
    limit: 10
  };

  onFetchRepos = () => this.props.fetchRepos(this.state.query, this.state.limit);

  onChange = e => this.setState({ query: e.target.value });

  render() {
    const { repos } = this.props;

    return (
      <div className="container">
        <div className="jumbotron jumbotron-fluid mt-5">
          <div className="container">
            <h1 className="display-4 text-warning">Repo searcher</h1>
            <p className="lead">Enter the topic that interests you.</p>
          </div>
        </div>

        <SearchForm
          onChange={this.onChange}
          fetchRepos={this.onFetchRepos}
        />
        <ReposList repos={repos} />
        <SavedRepos />
      </div>
    );
  }
};

export default connect(({ fetchedRepos }) => ({ repos: fetchedRepos }), { fetchRepos })(App);