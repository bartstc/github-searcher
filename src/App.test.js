import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import toJson from 'enzyme-to-json';

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(toJson(app)).toMatchSnapshot();
  });

  it('contains a SearchForm component', () => {
    expect(app.find('SearchForm').exists()).toBe(true);
  });

  it('contains a ReposList component', () => {
    expect(app.find('ReposList').exists()).toBe(true);
  });

  it('contains a SavedRepos component', () => {
    expect(app.find('Connect(SavedRepos)').exists()).toBe(true);
  });
});