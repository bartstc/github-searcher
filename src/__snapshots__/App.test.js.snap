// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron jumbotron-fluid mt-5"
  >
    <div
      className="container"
    >
      <h1
        className="display-4 text-warning"
      >
        Repo searcher
      </h1>
      <p
        className="lead"
      >
        Enter the topic that interests you.
      </p>
    </div>
  </div>
  <SearchForm
    fetchRepos={[Function]}
    onChange={[Function]}
  />
  <ReposList />
  <Connect(SavedRepos) />
</div>
`;
